### PR TITLE
[JS] feat: add teams sso message extension sample

### PR DESCRIPTION
**Pull request by blackchoey**
## Linked issues

closes: #835, #782

## Details

Add Teams SSO message extension sample, both local debug and remote deploy is ready.
When user types `profile` in Message Extension's search box, it will show current user's profile.

Note:
Like other new samples, since the latest code is not published, people need to update package.json manually to reference a local package before try this sample.
 
## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by singhk97 — APPROVED** [review]

File: js/samples/06.auth.teamsSSO.messagingExtension/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    ```
+
+1. [Upload the app](https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/deploy-and-publish/apps-upload) file (manifest.zip created in the previous step) in Teams.
+
+
+## Deploy the bot to Azure
+
+You can use Teams Toolkit for VS Code or CLI to host the bot in Azure. The sample includes Bicep templates in the `/infra` directory which are used by the tools to create resources in Azure. Use the **Provision** and **Deploy** menus of the Teams Toolkit extension or from the CLI with `teamsfx provision` and `teamsfx deploy`. [Visit the documentation](https://learn.microsoft.com/en-us/microsoftteams/platform/toolkit/provision) for more info on hosting your app in Azure with Teams Toolkit.
+
+Alternatively, you can learn more about deploying a bot to Azure manually in the [Deploy your bot to Azure](https://aka.ms/azuredeployment) documentation.
```
Comment (corinagum) [inline js/samples/06.auth.teamsSSO.messagingExtension/README.md:102]: Could you update these sections to match echobot's? Thanks!

https://github.com/microsoft/teams-ai/blob/main/js/samples/01.messaging.a.echoBot/README.md#teams-toolkit-for-visual-studio-code

**Comment by corinagum** [comment]
Could you also rename the directory of the same to be `messageExtension` or `messageExtensions` instead of `messagingExtension`?

**Review by corinagum — CHANGES_REQUESTED** [review]
2 nitpicks. Please ping me when done and I can approve asap~

File: js/samples/06.auth.teamsSSO.messagingExtension/README.md (hunk 1 above)
Comment (blackchoey) [reply js/samples/06.auth.teamsSSO.messagingExtension/README.md]: Thanks, updated. I removed the bot emulator part as it should not support Message Extension functionality.

**Comment by blackchoey** [comment]
> Could you also rename the directory of the same to be `messageExtension` or `messageExtensions` instead of `messagingExtension`?

Renamed to `messageExtension` to follow JS SDK's naming guideline.

**Review by corinagum — APPROVED** [review]

**Pull request merged**